File: Balanced-Binary-Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool ok = 1;
    int dfs(TreeNode *root){
        if(!root)
            return 0;
        int dl = dfs(root->left);
        int dr = dfs(root->right);
        int d = max(dr, dl) + 1;
        if(abs(dr - dl) > 1){
            ok =0;
            return 0;
        }
        return d;
    }
    bool isBalanced(TreeNode* root) {
        dfs(root);
        return ok;
    }
};

File: Balanced-Binary-Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */

var isBalanced = function(root) {
     let ok = true;

    function dfs(node) {
        if (!node) return 0;

        let dl = dfs(node.left);
        let dr = dfs(node.right);

        if (Math.abs(dl - dr) > 1) {
            ok = false;
            return 0;
        }

        return Math.max(dl, dr) + 1;
    }

    dfs(root);
    return ok;
};